[FEAT] Reject Adhoc Messages

afallenhope opened the pull request:
This PR is to address an annoyance where users automatically add you to a conference.
As a result, I've added the option to automatically leave the conference (adhoc), and announce (if requested). 
Idea came from the very talented Firestorm folks so I appreciate their take on this. ~I've bound the event on `llagent`.~

*Known Issues / Caveats
None that I'm aware of

afallenhope committed:
FLN-reject-adhoc xml changes for settings and preferences

afallenhope committed:
FLN-reject-adhoc logic code changes

cinderblocks commented:
@PanteraPolnocy 's implementation is less complicated and functionally does the same thing. Why not just use that? I don't see the advantage in tying gAgent in when gIMMgr is also global.

Also, having two conflicting boolean settings can be enumerated under one setting. Off, Friends, All. 

"Reports to nearby chat if a conference (ad-hoc) has been ignored." is also very Firestorm-specific wording for a system message. I don't see why it would logged to nearby chat when a session id exists either. Makes for messy chat logs.

afallenhope commented:
> @PanteraPolnocy 's implementation is less complicated and functionally does the same thing. Why not just use that? I don't see the advantage in tying gAgent in when gIMMgr is also global.
> 
> Also, having two conflicting boolean settings can be enumerated under one setting. Off, Friends, All.
> 
> "Reports to nearby chat if a conference (ad-hoc) has been ignored." is also very Firestorm-specific wording for a system message. I don't see why it would logged to nearby chat when a session id exists either. Makes for messy chat logs.

Thanks @cinderblocks I appreciate the feedback.

- I chose to use `gAgent` because I noticed the `isDoNotDisturb` setting is already tied to it, and it seemed to align well with the functionality was being implemented. 

- I also appreciate the idea of combining the two conflicting booleans into a single enumerated setting. The proposed values could be something like:

```
<combo_box
        control_name="AlchemyIgnoreAdHocSessions"
        follows="left|top"
        layout="topleft"
        height="23"
        name="adhoc_preference_combo"
        width="115">
    <item
        label="Off"
        name="adhoc_preference_allow_none"
        value="0"/>
   <item
        label="Friends Only"
        name="adhoc_preference_allow_friends"
        value="1"/>
    <item
        label="All"
        name="adhoc_preference_allow_all"
        value="2"/>
  </combo_box>
```

This would make it more flexible.
Changing from `getBOOL` to `getS32`. 
UI wise, a checkbox instead of dropdowns was much easier (for myself). 

I thank you again for the valued feedback.

PanteraPolnocy commented:
> "Reports to nearby chat if a conference (ad-hoc) has been ignored." is also very Firestorm-specific wording for a system message. I don't see why it would logged to nearby chat when a session id exists either. Makes for messy chat logs.

This was implemented because some users would forget that ad-hoc sessions had been ignored, leading to confusion, so the message to nearby serves as a visible reminder.

afallenhope commented:
> > "Reports to nearby chat if a conference (ad-hoc) has been ignored." is also very Firestorm-specific wording for a system message. I don't see why it would logged to nearby chat when a session id exists either. Makes for messy chat logs.
> 
> This was implemented because some users would forget that ad-hoc sessions had been ignored, leading to confusion, so the message to nearby serves as a visible reminder.

Personally I think it's a great reminder. Thank you for your great work on this.
I did take the suggestions that @cinderblocks has recommended.

1. Removed it from `llagent` and strictly read it from the `gSettingsPerAccount.getS32("AlchemyIgnoreAdHocSessions")` 
2. Changed to a combobox which does look nicer instead of two checkboxes
  ![image](https://github.com/user-attachments/assets/7768fcde-7e5d-4b78-ae93-e07a2d7693f3)

afallenhope committed:
Removed binding to agent

afallenhope committed:
simplified as per PR request and add new combo box

afallenhope committed:
missed a spot

afallenhope committed:
PR suggestion to use notifications vs system messages

afallenhope commented:
>  I see you already made the change.

Yes, it was a great  suggestion and I value the feeback.

> Case could be made that DoNotDisturb and AutoRespond have no place in gAgent either as they are instant messaging states and not agent control states

I feel that would have to be in a separate PR as that would relate to a separate request.

I've also took the suggestion of using `LLNotificationsUtil` to display the notification. Thank you for the valued feedback.
